draft-edm-protocol-greasing.md

```diff
@@ -156,6 +156,35 @@ cause a new class of interoperability failure. Therefore a label such as
 "reserved for greasing" may help to avoid the confusion.
 
 
+# Deployment Considerations and Incentives for Greasing
+
+Greasing can be used as a tool to improve the active use of existing protocol
+elements (which weren't necessarily designed with greasing to begin with, or
+weren't deployed with greasing); or as part of new protocol design and deployments.
+
+When greasing isn't used from the beginning of protocol deployment, starting to use
+greasing comes with the risk of triggering failures. These failures might be innocuous,
+but they also might be very impactful and visible to users. This risk creates a
+disincentive to deploy greasing in existing systems, since generally the change that
+triggers failures is often blamed for the failure.
+
+Some approaches to avoid failures due to greasing include:
+
+- Designing, implementing, and using greasing very early on in protocol development
+and deployment. This avoids the risks of adding greasing later.
+- Enabling greasing along with other major protocol feature changes or deployment changes.
+For example, when upgrading to a new protocol version that requires implementation updates
+on multiple systems, greasing can be added for the new version specifically. This approach
+works well for situations where the protocol participants are known and already need
+to cooperate (such as within an encrypted protocol between two endpoints). This approach
+applies less well for situations where non-cooperating entities (such as middleboxes)
+are the source of ossification.
+- Using heuristics to disable greasing when errors are encountered. For example,
```

Comment:
I read this a couple of times and I'm not sure if the intention is to describe dynamic behaviour using heuristics specifically when errors are encounterd, or if static methods / pre-empting failue are covered too.

In other words, I might know apriori that a subset of clients have a known issue with greasing without needing   to encounter the problem myself. One approach to avoiding the problem is to partition the "existing system" based on something like a client identifer (ASN, versions exchanged in a handshake prior to when problematic greasing might be encountered, etc.) and special case them for a time period (which might extend to infinity).

Reply:
Added text to explain a server-view of this!